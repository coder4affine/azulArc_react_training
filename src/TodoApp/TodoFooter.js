import React, { memo } from 'react';
import PropTypes from 'prop-types';

const TodoFooter = ({ switchList }) => {
  console.log('TodoFooter');
  return (
    <div>
      <button type="button" onClick={() => switchList('all')}>
        All Todos
      </button>
      <button type="button" onClick={() => switchList('pending')}>
        Pending Todos
      </button>
      <button type="button" onClick={() => switchList('completed')}>
        Completed Todos
      </button>
    </div>
  );
};

TodoFooter.propTypes = {
  switchList: PropTypes.func.isRequired,
};

export default memo(TodoFooter);
