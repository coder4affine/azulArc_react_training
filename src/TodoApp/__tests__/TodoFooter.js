import React from 'react';

import { shallow } from 'enzyme';
import TodoFooter from '../TodoFooter';

function setup() {
  const props = {
    switchList: jest.fn(),
  };
  const wrapper = shallow(<TodoFooter {...props} />);
  return { wrapper, props };
}

describe('test TodoFooter Component', () => {
  it('check div exist', () => {
    const { wrapper } = setup();
    const div = wrapper.find('div');
    expect(div.exists()).toBe(true);
    expect(div.length).toBe(1);
  });

  it('check count button exist and check count', () => {
    const { wrapper } = setup();
    const button = wrapper.find('button');
    expect(button.exists()).toBe(true);
    expect(button.length).toBe(3);
  });

  it('check text of button', () => {
    const { wrapper } = setup();
    const button = wrapper.find('button');
    expect(button.length).toBe(3);
    expect(button.at(0).text()).toEqual('All Todos');
    expect(button.at(1).text()).toEqual('Pending Todos');
    expect(button.at(2).text()).toEqual('Completed Todos');
  });

  it('check click event working', () => {
    const { wrapper, props } = setup();
    const button = wrapper.find('button');
    expect(button.length).toBe(3);
    button.at(0).simulate('click');
    button.at(1).simulate('click');
    button.at(2).simulate('click');
    expect(props.switchList.mock.calls.length).toBe(3);
  });
});
