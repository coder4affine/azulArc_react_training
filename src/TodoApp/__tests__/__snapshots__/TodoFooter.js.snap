// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TodoFooter correctly 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    All Todos
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Pending Todos
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Completed Todoss
  </button>
</div>
`;
